datafusion/physical-plan/src/joins/sort_merge_join.rs

```diff
@@ -1608,8 +1619,6 @@ fn get_buffered_columns_from_batch(
 
             Ok(buffered_cols)
         }
-        // Invalid combination
```

Comment:
The compiler now can validate that that this combination is invalid rather than relying on a runtime check